docs/specifications/tests/MethodsV2.md

```diff
@@ -0,0 +1,952 @@
+  least one [valid name server name] with at least one [valid IP address].
+* "Undelegated Data" - Optonal data. If included it must consist of a set of
+  at least one [valid name server name] and for each name server name an optional
+  set of at least one [valid IP address]. The name servers and IP addresses
+  represent a possible delegation of *Child Zone* from its parent zone (may be
+  indetermined).
+* "Test Type" - Derived data. It is set to "normal test" if *Undelegated Data* is
+  absent (empty) and to "undelegated test" if it is non-empty.
+
+[To top]
+
+
+## Method: Get parent NS IP addresses
+
+### Method identifier
+**Get-Parent-NS-IP**
+
+### Objective
+
+This Method will obtain the name servers that serves the parent zone, i.e. the
+zone from which the *Child Zone* is delegated from.
+
+This is done by finding the parent zone and then the name servers that serve the
+parent zone. In case there is an inconsistency of which is the parent zone, the
+list of name servers will be the gross list, i.e. rather include too much than
+too little. Too much is always a result of incorrect configuration in the parent
+zone or in a grand parent zone.
+
+If *Child Zone* is the root zone, then there is by definition no parent zone and
+no parent name servers.
+
+If the test type is undelegated, then the information that the parent name
+servers are supposed to provide included in the input data. In that case a list
+of parent name servers has no meaning.
+
+The Method will output a list of parent name server IP addresses. If the zone is
+the root zone or if the test is an undelegated test, the list is defined but
+empty. If the parent zone cannot be determined, then an undefined list is
+returned.
+
+This Method must, in general, use the similar algorithm as Test Case [BASIC01],
+but the test case extracts more information.
+
+### Inputs
+
+This Method uses the following input units defined in section [Methods Inputs]:
+
+* "Child Zone" - The name of the child zone to be tested.
+* "Root Name Servers"
+* "Test Type" - "undelegated test" or "normal test".
+
+
+### Procedures
+
+1. If the *Child Zone* is the root zone (".") then output empty set and exit
+   these procedures.
+
+2. If the *Test Type* is "undelegated test" then output empty set and exit
+   these procedures.
+
+3. Create a [DNS Query] with query type SOA and query name *Child Zone*
+   ("SOA Child Query").
+
+4. Create the following empty set:
+   1. Parent name server IP and the parent zone name ("Parent Name Server IP").
+
+5. Find the parent zone of the *Child Zone* by iteratively [sending][send]
+   *SOA Child Query* to all name servers found. Start by using the nameservers
+   from *Root Name Servers*.
+   1. Follow all paths from root and downwards by using the referrals (non-AA
+      response with empty answer section and NS records in the authority
+      section).
+   2. When one of the following criteria is met (not both), then stop the lookup
```

Comment:
When exactly one of the following criteria is met, then stop the lookup…

Reply:
The two cannot be true at the same time if the response is valid. I added a term "Referral" in the terminology section and rewrote the bullet above. I also updated in several places to use the term I defined.